### PR TITLE
Zperf multicast improvements

### PR DESCRIPTION
**Pull request by krish2718**
Multicast receive for Zperf.

### CODE REVIEW

File: subsys/net/lib/zperf/zperf_shell.c [hunk 1]
```diff
@@ -191,7 +191,7 @@ static int cmd_setip(const struct shell *sh, size_t argc, char *argv[])
 		if (zperf_get_ipv6_addr(argv[start + 1], argv[start + 2],
 					&shell_ipv6) < 0) {
 			shell_fprintf(sh, SHELL_WARNING,
-				      "Unable to set IP\n");
+				      "Unable to set %s address (%s disabled)\n", "IPv6", "IPv4");
```
Comment (rado17) [inline subsys/net/lib/zperf/zperf_shell.c:227]: Why pass const string as argument instead having it as part of the fmt itself?

File: subsys/net/lib/zperf/zperf_shell.c [hunk 2]
```diff
@@ -1110,7 +1110,7 @@ void zperf_shell_init(void)
 {
 	int ret;
 
-	if (IS_ENABLED(CONFIG_NET_IPV6) && MY_IP6ADDR) {
+	if (IS_ENABLED(MY_IP4ADDR_SET) && MY_IP6ADDR) {
```
Comment (rado17) [inline subsys/net/lib/zperf/zperf_shell.c:1150]: Should this be IS_ENABLED(MY_IP6ADDR_SET)

File: subsys/net/lib/zperf/zperf_shell.c (hunk 1 above)
Comment (krish2718) [reply subsys/net/lib/zperf/zperf_shell.c:227]: To save memory, the format strings are RO and can be re-used if they are same saving memory.

File: subsys/net/lib/zperf/zperf_shell.c (hunk 2 above)
Comment (krish2718) [reply subsys/net/lib/zperf/zperf_shell.c:1150]: oops, need to fix upstream as well, good catch.

Comment (krish2718) [reply subsys/net/lib/zperf/zperf_shell.c]: updated with the fix.

**Review by rado17 — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

**Pull request merged**